apps/interactive-calibration/calibPipeline.cpp

```diff
@@ -29,37 +29,61 @@ CalibPipeline::CalibPipeline(captureParameters params) :
 {
-    if(mCaptureParams.source == Camera && !mCapture.isOpened())
-    {
-        mCapture.open(mCaptureParams.camID);
-        cv::Size maxRes = getCameraResolution();
-        cv::Size neededRes = mCaptureParams.cameraResolution;
-
-        if(maxRes.width < neededRes.width) {
-            double aR = (double)maxRes.width / maxRes.height;
-            mCapture.set(cv::CAP_PROP_FRAME_WIDTH, neededRes.width);
-            mCapture.set(cv::CAP_PROP_FRAME_HEIGHT, neededRes.width/aR);
+    auto open_camera = [this] () {
+        if(mCaptureParams.source == Camera)
+        {
+            mCapture.open(mCaptureParams.camID);
+            cv::Size maxRes = getCameraResolution();
+            cv::Size neededRes = mCaptureParams.cameraResolution;
+
+            if(maxRes.width < neededRes.width) {
+                double aR = (double)maxRes.width / maxRes.height;
+                mCapture.set(cv::CAP_PROP_FRAME_WIDTH, neededRes.width);
+                mCapture.set(cv::CAP_PROP_FRAME_HEIGHT, neededRes.width/aR);
+            }
+            else if(maxRes.height < neededRes.height) {
+                double aR = (double)maxRes.width / maxRes.height;
+                mCapture.set(cv::CAP_PROP_FRAME_HEIGHT, neededRes.height);
+                mCapture.set(cv::CAP_PROP_FRAME_WIDTH, neededRes.height*aR);
+            }
+            else {
+                mCapture.set(cv::CAP_PROP_FRAME_HEIGHT, neededRes.height);
+                mCapture.set(cv::CAP_PROP_FRAME_WIDTH, neededRes.width);
+            }
+            mCapture.set(cv::CAP_PROP_AUTOFOCUS, 0);
         }
-        else if(maxRes.height < neededRes.height) {
-            double aR = (double)maxRes.width / maxRes.height;
-            mCapture.set(cv::CAP_PROP_FRAME_HEIGHT, neededRes.height);
-            mCapture.set(cv::CAP_PROP_FRAME_WIDTH, neededRes.height*aR);
-        }
-        else {
-            mCapture.set(cv::CAP_PROP_FRAME_HEIGHT, neededRes.height);
-            mCapture.set(cv::CAP_PROP_FRAME_WIDTH, neededRes.width);
-        }
-        mCapture.set(cv::CAP_PROP_AUTOFOCUS, 0);
+        else if (mCaptureParams.source == File)
+            mCapture.open(mCaptureParams.videoFileName);
+    };
+
+    if(!mCapture.isOpened()) {
+        open_camera();
     }
-    else if (mCaptureParams.source == File && !mCapture.isOpened())
-        mCapture.open(mCaptureParams.videoFileName);
     mImageSize = cv::Size((int)mCapture.get(cv::CAP_PROP_FRAME_WIDTH), (int)mCapture.get(cv::CAP_PROP_FRAME_HEIGHT));
 
     if(!mCapture.isOpened())
         throw std::runtime_error("Unable to open video source");
 
+    bool checkImageSize = false;
     cv::Mat frame, processedFrame;
-    while(mCapture.grab()) {
+    while (true) {
+        bool ret = mCapture.grab();
+        if (!ret)
+        {
+            if (!mCaptureParams.forceReopen)
+            {
+                break;
+            }
+            open_camera();
+            checkImageSize = true;
+            continue;
+        }
```

Comment:
Some console logging/messages would be useful for OpenCV users. It makes sense to mention parameter hint too.